primitives/nomad/merkle/src/lib.rs

```diff
@@ -31,20 +31,112 @@ pub type NomadLightMerkle = light::LightMerkle<TREE_DEPTH>;
 pub type NomadProof = proof::Proof<TREE_DEPTH>;
 
 pub use error::{TreeError, VerifyingError, VerifyingError::VerificationFailed};
+use hex_literal::hex;
 pub use light::*;
 pub use proof::*;
 pub use utils::*;
 
-lazy_static::lazy_static! {
-	/// A cache of the zero hashes for each layer of the tree.
-	pub static ref ZERO_HASHES: [H256; TREE_DEPTH + 1] = {
-		let mut hashes = [H256::zero(); TREE_DEPTH + 1];
-		for i in 0..TREE_DEPTH {
-			hashes[i + 1] = hash_concat(hashes[i], hashes[i]);
-		}
-		hashes
-	};
-}
+/// A cache of the zero hashes for each layer of the tree.
+/// See `zero_hashes_pre_build()` test to check how it is calculated.
+pub const ZERO_HASHES: [H256; TREE_DEPTH + 1] = [
```

Comment:
Why we want to avoid lazy_static, is that something we should avoid? I'm not sure if it makes sense to change `TREE_DEPTH`, but in previous implementation you could change it without causing compilation to fail.

Reply:
The WASM interpreter could be re-loaded several times (i.e. after a `set_code` tx), so next call to `ZERO_HASHES` could spend more time than expected (a.k.a. calculated by benchmark) and the block execution could be out of time (the worst case), producing forks.

That situation will be really difficult to track/debug. So the idea is to replace any calculation at runtime by its output at compile time, specially if it is long time operations.

In order to warranty that pre-image contains the proper calculated values, we have the test case `zero_hashes_pre_build()`.

Reply:
Oh, I didn't considered WASM runtime. Sounds like nice bug to have, though :D